### ISSUE TITLE
How to decrease the max number of worker during training

### ISSUE THREAD

**Issue by biocaitao** [slideflow/slideflow#396]
## Description
I installed the slideflow on a cluster using the "Slurm" job system in that we are limited to 1 node with 1 A100 GPU.  
When testing the demo cases of sliderflow:  
params = sf.ModelParams(
    tile_px=299,
    tile_um='5x',
    model='xception',
    batch_size=32,
    learning_rate=0.0001,
    epochs=[10]
)
labels, _ = dataset.labels('category')
trainer = sf.model.build_trainer(
    hp=params,
    outdir='./sfdata/trainer',
    labels=labels
)
results = trainer.train(train_dataset, test_dataset)

I got the warning like below: 
UserWarning: This DataLoader will create 4 worker process
es in total. Our suggested max number of worker in current system is 2, which is
 smaller than what this DataLoader is going to create. Please be aware that exce
ssive worker creation might get DataLoader running slow or even freeze, lower th
e worker number to avoid potential slowness/freeze if necessary.

So, how to limit the number of workers during training?

Thanks.

**Comment by jamesdolezal** [comment]
Hi biocaitao,

At present, the number of workers is hard-coded to 4. However, we have just added an update (available on our `dev` branch), which will allow users to customize the number of dataloader workers. An example of how to use:

```python
trainer = sf.model.build_trainer(..., num_workers=2)
```
or
```python
trainer = sf.model.build_trainer(...)
trainer.num_workers = 2
```
We will include this feature in version 2.4.0.

**Comment by jamesdolezal** [comment]
Included in version 3.0
